beacon-chain/core/blocks/block_operations.go

```diff
@@ -97,30 +97,51 @@ func verifyDepositDataWithDomain(ctx context.Context, deps []*ethpb.Deposit, dom
 		}
 		msgs[i] = ctrRoot
 	}
-	as := bls.AggregateSignatures(sigs)
-	if !as.AggregateVerify(pks, msgs) {
-		return errors.New("one or more deposit data signatures did not verify")
+	verify, err := bls.VerifyMultipleSignatures(sigs, msgs, pks)
+	if err != nil {
+		return errors.Errorf("got error in multiple verification: %v", err)
+	}
+	if !verify {
+		return errors.New("one or more deposit signatures did not verify")
 	}
 	return nil
 }
 
-func verifySignature(signedData []byte, pub []byte, signature []byte, domain []byte) error {
+// retrieves the signature set from the raw data, public key,signature and domain provided.
+func retrieveSignatureSet(signedData []byte, pub []byte, signature []byte, domain []byte) (*bls.SignatureSet, error) {
 	publicKey, err := bls.PublicKeyFromBytes(pub)
 	if err != nil {
-		return errors.Wrap(err, "could not convert bytes to public key")
+		return nil, errors.Wrap(err, "could not convert bytes to public key")
 	}
 	sig, err := bls.SignatureFromBytes(signature)
 	if err != nil {
-		return errors.Wrap(err, "could not convert bytes to signature")
+		return nil, errors.Wrap(err, "could not convert bytes to signature")
 	}
 	signingData := &pb.SigningData{
 		ObjectRoot: signedData,
 		Domain:     domain,
 	}
 	root, err := ssz.HashTreeRoot(signingData)
 	if err != nil {
-		return errors.Wrap(err, "could not hash container")
+		return nil, errors.Wrap(err, "could not hash container")
 	}
+	return &bls.SignatureSet{
+		Signatures: []bls.Signature{sig},
+		PublicKeys: []bls.PublicKey{publicKey},
+		Messages:   [][32]byte{root},
+	}, nil
+}
+
+// verifies the signature  from the raw data, public key and domain provided.
+func verifySignature(signedData []byte, pub []byte, signature []byte, domain []byte) error {
+	set, err := retrieveSignatureSet(signedData, pub, signature, domain)
+	if err != nil {
+		return err
+	}
+	// We assume only one signature set is returned here.
+	sig := set.Signatures[0]
```

Comment:
Instead of assuming, would it be better to error if more than one signature set is returned? I think that makes me feel better

Error if set.Signatures[0] != set.Signatures[1]... etc